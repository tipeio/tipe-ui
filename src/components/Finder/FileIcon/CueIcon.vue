<template>
  <button :data-tipe-ui="$options.name">
    <tipe-icon
      icon="options-horizontal"
      height="0.6875rem"
      width="0.6875rem"
    />
  </button>
</template>

<script>
import TipeIcon from '@/components/Icon'

export default {
  name: 'TipeCueIcon',
  components: {
    TipeIcon
  }
}
</script>


<style lang="postcss" scoped>
[data-tipe-ui='TipeCueIcon'] {
  height: 1.5625rem;
  width: 1.5625rem;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid var(--gray);
  border-radius: 100%;
  background-color: #fff;
  box-shadow: 0 0.25rem 0.875rem 0 rgba(0, 0, 0, 0.1);
  padding: 0;
  margin: 0;
  overflow: visible;
}
</style>
