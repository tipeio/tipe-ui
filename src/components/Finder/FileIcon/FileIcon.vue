<template>
  <tipe-file-icon-rectangle
    v-if="type == 'rectangle'"
    :file="file"
  />
  <tipe-file-icon-square
    v-else-if="type == 'square'"
    :file="file"
  />
</template>


<script>
import vueTypes from 'vue-types'
import TipeFileIconRectangle from './FileIconRectangle'
import TipeFileIconSquare from './FileIconSquare'

export default {
  name: 'FileIcon',
  components: {
    TipeFileIconRectangle,
    TipeFileIconSquare
  },
  props: {
    type: vueTypes.oneOf(['rectangle', 'square']).def('rectangle'),
    file: vueTypes.shape({
      type: vueTypes.string,
      label: vueTypes.string
    })
  }
}
</script>
