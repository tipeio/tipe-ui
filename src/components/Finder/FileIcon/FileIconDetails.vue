<template>
  <div
    :data-tipe-ui="$options.name"
    :class="rootClassObject"
    @mouseover="hovered = true"
    @mouseout="hovered = false"
    @click="$emit('click', $event)"
  >
    <button
      class="details-button"
      @click="$emit('click')"
    >
      <div class="details-layout">
        <div class="icon-container">
          <tipe-icon :icon="icon"/>
        </div>
        <div class="label">
          {{ label }}
        </div>
        <div class="created-by">
          Created by {{ author }}
        </div>
        <div class="dot">&#8226;</div>
        <div class="updated-at">
          Edited {{ updatedAt }}
        </div>
      </div>
    </button>
    <button
      v-if="hovered && !picker"
      class="trash"
      @click="$emit('delete')"
    >
      <tipe-icon icon="trash"/>
    </button>
    <button
      v-if="hovered && !picker"
      class="edit"
      @click="$emit('edit')"
    >
      <tipe-icon icon="edit"/>
    </button>
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import TipeIcon from '@/components/Icon'
import interfaces from '@tipe/tipe-interfaces'

export default {
  name: 'TipeFileIconDetails',
  components: {
    TipeIcon
  },
  props: {
    picker: vueTypes.bool.def(false),
    selected: vueTypes.bool.def(false),
    ...interfaces.file
  },
  data() {
    return {
      hovered: false
    }
  },
  computed: {
    author() {
      return `${this.createdBy.firstName} ${this.createdBy.lastName}`
    },
    rootClassObject: props => ({
      selected: props.selected
    })
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeFileIconDetails'] {
  display: grid;
  grid-template-columns: 1fr 3rem 3rem;
  grid-template-rows: 4.8125rem;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 1px solid var(--gray-light);
}

[data-tipe-ui='TipeFileIconDetails']:first-child {
  border-top: 1px solid var(--gray-light);
}

[data-tipe-ui='TipeFileIconDetails']:nth-child(odd) {
  background-color: rgba(0, 0, 0, 0.01);
}

[data-tipe-ui='TipeFileIconDetails']:hover {
  background-color: rgba(255, 255, 255, 0.9);
}

[data-tipe-ui='TipeFileIconDetails'].selected {
  background-color: rgba(0, 0, 0, 0.06);
}

.details-layout {
  display: grid;
  grid-template-areas: 'icon label label label' 'icon created-by dot updated-at';
  grid-template-columns: 3rem auto auto auto auto;
  grid-template-rows: 1fr 1fr;
  overflow: hidden;
  justify-content: flex-start;
}

.icon-container {
  grid-area: icon;
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.label {
  grid-area: label;
  grid-column: 1fr;
  display: grid;
  white-space: nowrap;
  font-size: 0.875rem;
  white-space: nowrap;
  letter-spacing: 0.008125rem;
  color: var(--text-gray);
  font-weight: 500;
  align-items: center;
  justify-content: flex-start;
}

.created-by {
  grid-area: created-by;
  grid-column: auto;
  display: grid;
  align-items: center;
  justify-content: flex-start;
  font-size: 0.75rem;
  white-space: nowrap;
  letter-spacing: 0.006875rem;
  color: var(--text-gray);
}

.updated-at {
  grid-area: updated-at;
  grid-column: 1fr;
  display: grid;
  align-items: center;
  justify-content: flex-start;
  font-size: 0.75rem;
  white-space: nowrap;
  letter-spacing: 0.006875rem;
  color: var(--text-gray);
}

.dot {
  grid-area: dot;
  color: var(--text-gray);
  font-size: 0.8rem;
  padding: 0 0.4rem;
}

button {
  margin: 0;
  padding: 0;
  border: none;
  outline: none;
  background-color: transparent;
  height: 100%;
  width: 100%;
}
</style>
