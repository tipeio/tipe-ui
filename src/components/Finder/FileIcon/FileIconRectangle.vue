<template>
  <div
    :data-tipe-ui="$options.name"
    :class="rootClassObject"
    @click="$emit('click', $event)"
  >
    <button
      class="file-icon"
      @click="$emit('click', $event)"
    >
      <tipe-icon :icon="icon" />
    </button>
    <button
      class="label"
      @click="$emit('click', $event)"
    >
      {{ label }}
    </button>
    <button
      v-if="!picker"
      class="options-icon"
    >
      <tipe-icon icon="options" />
    </button>
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import TipeIcon from '@/components/Icon'
import interfaces from '@tipe/tipe-interfaces'

export default {
  name: 'TipeFileIconRectangle',
  components: {
    TipeIcon
  },
  props: {
    picker: vueTypes.bool.def(false),
    selected: vueTypes.bool.def(false),
    ...interfaces.file
  },
  computed: {
    rootClassObject: props => ({
      selected: props.selected
    })
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeFileIconRectangle'] {
  height: 100%;
  width: 100%;
  min-width: 9rem;
  min-height: 3rem;
  display: grid;
  grid-template-columns: 3.75rem 1fr 3rem;
  border: 1px solid var(--gray-blue-light);
  border-radius: 0.3125rem;
  box-sizing: border-box;
}

.selected {
  border: 1px solid var(--text-gray-light);
}

button {
  border: none;
  outline: none;
  margin: 0;
  padding: 0;
  background-color: transparent;
}

.file-icon,
.options-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}

.label {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  white-space: nowrap;
  overflow: hidden;
  color: rgb(31, 52, 108);
  font-size: 0.8125rem;
}
</style>
