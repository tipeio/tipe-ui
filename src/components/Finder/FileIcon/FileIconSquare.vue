<template>
  <button
    :data-tipe-ui="$options.name"
    @click="$emit('click', $event)"
  >
    <div class="layout">
      <div class="file">
        <div class="file-content">
          <div class="header">
            <div class="label">
              {{ file.label }}
            </div>
          </div>
          <div class="icon-container">
            <tipe-icon
              icon="plus"
              height="0.6875rem"
              width="0.6875rem"
            />
          </div>
        </div>
        <div class="cue-container">
          <tipe-cue />
        </div>
      </div>
    </div>
  </button>
</template>

<script>
import vueTypes from 'vue-types'
import interfaces from '@tipe/tipe-interfaces'
import TipeIcon from '@/components/Icon'
import TipeCue from './CueIcon'

export default {
  name: 'TipeFileIconSquare',
  components: {
    TipeIcon,
    TipeCue
  },
  props: {
    file: vueTypes.shape(interfaces.file)
  },
  computed: {
    icon() {
      return {
        document: 'file',
        folder: 'folder'
      }[this.file.type]
    }
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeFileIconSquare'] {
  width: 100%;
  height: 100%;
  border: 0;
  outline: 0;
  margin: 0;
  padding: 0;
  background-color: transparent;
}

.layout {
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.file {
  position: relative;
  height: 9.625rem;
  width: 8.46875rem;
}

.file-content {
  height: 100%;
  width: calc(100% - 0.78125rem);
  display: grid;
  grid-template-rows: 2.1875rem 1fr;
  grid-auto-columns: 100%;
  overflow: hidden;
  border: 1px solid var(--gray-light);
  border-radius: 0.1875rem;
  box-sizing: border-box;
  box-shadow: 0 0.125rem 0.3125rem 0 rgba(0, 0, 0, 0.11);
}

.header {
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0 1rem;
  box-sizing: border-box;
  background-color: var(--purple);
}

.label {
  white-space: nowrap;
  color: #fff;
  font-size: 0.8125rem;
  font-weight: 500;
  overflow: hidden;
}

.icon-container {
  display: flex;
  align-items: center;
  justify-content: center;
}

.cue-container {
  position: absolute;
  bottom: 1rem;
  right: -0;
}
</style>
