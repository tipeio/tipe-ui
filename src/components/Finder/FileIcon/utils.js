const FileTypeToIconMap = {
  document: 'file',
  folder: 'folder',
  template: 'template'
}

const getFileIcon = file => FileTypeToIconMap[file.type]

export { getFileIcon }
