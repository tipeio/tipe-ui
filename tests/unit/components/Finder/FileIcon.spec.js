import { shallowMount } from '@vue/test-utils'
import FileIcon from '@/components/Finder/FileIcon'
import FileIconRectangle from '@/components/Finder/FileIcon/FileIconRectangle'
import FileIconSquare from '@/components/Finder/FileIcon/FileIconSquare'
import FileIconDetails from '@/components/Finder/FileIcon/FileIconDetails'
import { mocks, fixtures } from '@tipe/tipe-test-utils'
import tipeTestUtils from '~/tests/TipeTestUtils'

describe('FileIcon', () => {
  describe('<template>', () => {
    tipeTestUtils.test.snapshot(FileIcon, {
      propsData: { file: fixtures.file() }
    })
  })

  describe(':props', () => {
    it(':type(default) - should render FileIconRectangle component', () => {
      const propsData = { file: mocks.file() }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.contains(FileIconRectangle)).toEqual(true)
    })

    it(':type(square) - should render FileIconSquare component', () => {
      const propsData = { file: mocks.file(), type: 'square' }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.contains(FileIconSquare)).toEqual(true)
    })

    it(':type(default) - :file - should pass file to FileIconRectangle component', () => {
      const propsData = { type: 'rectangle', file: mocks.file() }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.find(FileIconRectangle).props()).toEqual(
        expect.objectContaining({
          label: propsData.file.label,
          icon: propsData.file.icon
        })
      )
    })

    it(':type(square) - :file - should pass file to FileIconSquare component', () => {
      const propsData = { type: 'square', file: mocks.file() }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.find(FileIconSquare).props()).toEqual(
        expect.objectContaining({
          label: propsData.file.label,
          icon: propsData.file.icon
        })
      )
    })

    it(':type(details) - :file - should pass file to FileIconDetails component', () => {
      const propsData = { type: 'details', file: mocks.file() }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.find(FileIconDetails).props()).toEqual(
        expect.objectContaining({
          label: propsData.file.label,
          icon: propsData.file.icon,
          createdBy: propsData.file.createdBy,
          updatedAt: propsData.file.updatedAt
        })
      )
    })

    it(':type(rectangle) - :picker - should pass picker to FileIconRectangle component', () => {
      const propsData = { type: 'rectangle', file: mocks.file(), picker: true }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.find(FileIconRectangle).props().picker).toEqual(true)
    })

    it(':type(square) - :picker - should pass picker to FileIconSquare component', () => {
      const propsData = { type: 'square', file: mocks.file(), picker: true }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.find(FileIconSquare).props().picker).toEqual(true)
    })

    it(':type(details) - :picker - should pass picker to FileIconDetails component', () => {
      const propsData = { type: 'details', file: mocks.file(), picker: true }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.find(FileIconDetails).props().picker).toEqual(true)
    })

    it(':type(rectangle) - :selected - should pass picker to FileIconRectangle component', () => {
      const propsData = {
        type: 'rectangle',
        file: mocks.file(),
        selected: true
      }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.find(FileIconRectangle).props().selected).toEqual(true)
    })

    it(':type(square) - :selected - should pass picker to FileIconSquare component', () => {
      const propsData = {
        type: 'square',
        file: mocks.file(),
        selected: true
      }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.find(FileIconSquare).props().selected).toEqual(true)
    })

    it(':type(details) - :selected - should pass picker to FileIconDetails component', () => {
      const propsData = {
        type: 'details',
        file: mocks.file(),
        selected: true
      }
      const wrapper = shallowMount(FileIcon, { propsData })

      expect(wrapper.find(FileIconDetails).props().selected).toEqual(true)
    })
  })
})
