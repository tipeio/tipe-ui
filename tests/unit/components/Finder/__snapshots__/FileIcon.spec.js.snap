// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileIcon <template> matches previous snapshot 1`] = `<TipeFileIconRectangle-stub id="" icon="test-icon" type="" label="test-label" createdAt="0" updatedAt="0"></TipeFileIconRectangle-stub>`;
